src/login/pam_systemd.c

```diff
@@ -997,6 +997,9 @@ _public_ PAM_EXTERN int pam_sm_open_session(
         desktop = getenv_harder(handle, "XDG_SESSION_DESKTOP", desktop_pam);
         incomplete = getenv_harder_bool(handle, "XDG_SESSION_INCOMPLETE", false);
 
+        if (streq_ptr(class, "none"))
+                goto success;
```

Comment:
i think this is too simple. 

we probably should still do the SHELL_PROMPT_PREFIX stuff here. 

(actually, the other relevant user of the "success" label below isn't right about this either, hmm, I need to clean that up).